src/light_data/light_store.rs

```diff
@@ -0,0 +1,405 @@
+            let de  = LightExpr::de(&ser).expect("read LightExpr");
+            assert_eq!(x, de)
+        }
+
+        #[test]
+        fn prop_light_store(s in any::<LightStore<Scalar>>()) {
+            let ser = s.ser();
+            let de  = LightStore::de(&ser).expect("read LightStore");
+            assert_eq!(s, de)
+        }
+
+        #[test]
+        fn test_convert_light_store_basic_strings((ptr3, ptr4, c1, c2) in any::<(ScalarPtr<Scalar>, ScalarPtr<Scalar>, char, char)>().prop_filter(
+            "Avoids confusion with StrNil",
+            |(ptr3, ptr4,c1, c2)| {
+                let strnil = ScalarPtr::from_parts(ExprTag::Str.as_field(), Scalar::zero());
+                *ptr3 != strnil && *ptr4 != strnil && *c2 != '\0' && *c1 != '\0'
+            })
+        ) {
+            let ptr1 = ScalarPtr::from_parts(ExprTag::Char.as_field(), Scalar::from_char(c1));
+            let ptr2 = ScalarPtr::from_parts(ExprTag::Char.as_field(), Scalar::from_char(c2));
+
+            let mut store = BTreeMap::new();
+            let strnil = ScalarPtr::from_parts(ExprTag::Str.as_field(), Scalar::zero());
+            store.insert(ptr3, Some(LightExpr::StrCons(ptr1, strnil)));
+            store.insert(ptr4, Some(LightExpr::StrCons(ptr2, ptr3)));
+
+            let expected_output = {
+                let mut output = ScalarStore::default();
+                output.insert_scalar_expression(
+                    ScalarPtr::from_parts(ExprTag::Str.as_field(), Scalar::zero()),
+                    Some(ScalarExpression::Str(String::from(""))),
+                );
+                output.insert_scalar_expression(ptr1, Some(ScalarExpression::Char(c1)));
+                output.insert_scalar_expression(ptr2, Some(ScalarExpression::Char(c2)));
+                output.insert_scalar_expression(ptr3, Some(ScalarExpression::Str(c1.to_string())));
+                output.insert_scalar_expression(ptr4, Some(ScalarExpression::Str(c2.to_string() + &c1.to_string())));
+
+                output
+            };
+
+            assert_eq!(LightStore::to_scalar_store(&LightStore{scalar_map: store}).unwrap(), expected_output);
+        }
+
+        #[test]
+        fn test_convert_light_store_basic_symbols((ptr3, ptr4, ptr5, ptr6, c1, c2) in any::<(ScalarPtr<Scalar>, ScalarPtr<Scalar>, ScalarPtr<Scalar>, ScalarPtr<Scalar>, char, char)>().prop_filter(
+            "Avoids confusion with StrNil, Symnil",
+            |(ptr3, ptr4, ptr5, ptr6, c1, c2)| {
+                let symnil = ScalarPtr::from_parts(ExprTag::Sym.as_field(), Scalar::zero());
+                let strnil = ScalarPtr::from_parts(ExprTag::Str.as_field(), Scalar::zero());
+                *ptr3 != strnil && *ptr4 != strnil && *ptr5 != strnil && *ptr6 != strnil &&
+                *ptr3 != symnil && *ptr4 != symnil && *ptr5 != symnil && *ptr6 != symnil &&
+                *c2 != '\0' && *c1 != '\0' &&
+                c2.to_string() != parser::SYM_SEPARATOR && c1.to_string() != parser::SYM_SEPARATOR
```

Comment:
I think we can have the sym separator inside symbols, like `.|foo.bar|` will have component `"foo.bar"`. If this isn't true in current lurk, we should make it so

Reply:
https://github.com/lurk-lang/lurk-rs/blob/932473fa129d459b5a671f105ffc38fd4602738a/src/sym.rs#L93

Please open an issue if you think a change is warranted.